astropy/cosmology/flrw/base.py

```diff
@@ -356,66 +312,77 @@ def m_nu(self, param, value):
     # properties
 
     @property
-    def is_flat(self):
+    def is_flat(self) -> bool:
         """Return bool; `True` if the cosmology is flat."""
-        return bool((self._Ok0 == 0.0) and (self.Otot0 == 1.0))
+        return bool((self.Ok0 == 0.0) and (self.Otot0 == 1.0))
 
     @property
-    def Otot0(self):
+    def Otot0(self) -> float:
```

Comment:
Is there a particular reason not to make this a `cached_property` as well?

Reply:
A good Q. Let's consider this in a followup PR! In this PR I was focused on getting rid of many of the ``object.__setattr__``.